src/modules/AdvancedPaste/AdvancedPaste/Helpers/JsonHelper.cs

```diff
@@ -149,7 +149,7 @@ internal static string ToJsonFromXmlOrCsv(DataPackageView clipboardData)
                         // A CSV line is valid, if the delimiter occurs more or equal times in every line compared to the first data line. (More because sometimes the delimiter occurs in a data string.)
                         if (line.Count(x => x == delim) >= delimCount)
                         {
-                            csv.Add(line.Split(delim));
+                            csv.Add(Regex.Split(line, delim + @"(?=(?:[^""]*""[^""]*"")*(?![^""]*""))"));
```

Comment:
It makes sense to move the regex as static readonly variable to the top. Line 23 to 25. 

And we should add a comment to describe how the regex works. Something like `"split on every occurence of the delim except if it is enclosed by " and ignore two " as escaped ".`".

Reply:
How could we do that? Since the regex string also consists of the variable `delim`.

Reply:
My bad. Forgot this.

Reply:
Like, we can do it but it won't make any sense. Also, instead of the variable being a `Regex`, it can be a `string`.

Reply:
I think the only benefit would that we don't copy the regex code into multiple places. Because this happens when you do the other points I mentioned in the main discussion of this PR. But it feels a bit strange to split the regex into two parts.

I think only the comment describing the regex makes sense to add. The variable thing makes no sense.

Reply:
I added the variable `CsvDelimSepRegexStr` to signify the body of the Regex, as it was being used in different places. I also added a comment explaining its usage.